### PR TITLE
Standardize Python file headers for copyright and license information

### LINKED ISSUE #4117: Get rid of `# Modified by:`, `# Author:`, `# Copyright <date>`, etc. in script headers
Ever since SCT's beginning, CLI script authors have followed a template for the header of each script, containing lines referencing authors/dates. 

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/822d1a34bd672d288ccf6c56316833f030a195fa/spinalcordtoolbox/scripts/sct_analyze_lesion.py#L5-L7

These lines tend to raise questions when making updates (https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4112#discussion_r1195012975), as we don't really have a protocol for updating dates/authors.

Additionally, because we have git, the record of when and who modified a file is already present.

So, to avoid future confusion, we could do a full sweep of all CLI scripts to remove author/date information from each script. (Question: Which information to remove, and which to keep?)

### PR DESCRIPTION
**Pull request by mguaypaq**
Fixes #4117. Sorry for the massive commit, but I don't think there's really a better way to do it.

Scripts with a shebang line now use the following header format, in comments:
```py3
#!/usr/bin/env python
#
# <DESCRIPTION>
#
# Copyright (c) <YEAR> Polytechnique Montreal <www.neuro.polymtl.ca>
# License: see the file LICENSE
```

Importable modules use the following format, in their docstring:
```py3
"""
<DESCRIPTION>

Copyright (c) <YEAR> Polytechnique Montreal <www.neuro.polymtl.ca>
License: see the file LICENSE
"""
```

In particular, the first copyright year and a pointer to the license file are kept, but author/last-modified information is removed.

When the copyright year was missing, I guessed from the git history.

### CODE REVIEW

File: spinalcordtoolbox/deepseg_/gm.py [hunk 1]
```diff
@@ -1,10 +1,15 @@
-# This is the interface API for the deepseg_gm model
-# that implements the model for the Spinal Cord Gray Matter Segmentation.
-#
-# Reference paper:
-#     Perone, C. S., Calabrese, E., & Cohen-Adad, J. (2017).
-#     Spinal cord gray matter segmentation using deep dilated convolutions.
-#     URL: https://arxiv.org/abs/1710.01269
+"""
+API for the deepseg_gm model that implements the model for the
+Spinal Cord Gray Matter Segmentation
```
Comment (joshuacwnewton) [inline spinalcordtoolbox/deepseg_/gm.py:3]: Tiny nit, but could we standardize the strings of the three `deepseg/_module.py` docstrings?

```suggestion
API for functions used in the sct_deepseg_gm CLI script
```

File: spinalcordtoolbox/deepseg_/lesion.py [hunk 2]
```diff
@@ -1,4 +1,9 @@
-# Functions dealing with deepseg_lesion
+"""
+Functions dealing with deepseg_lesion
```
Comment (joshuacwnewton) [inline spinalcordtoolbox/deepseg_/lesion.py:2]:
```suggestion
API for functions used in the sct_deepseg_lesion CLI script
```

**Review by joshuacwnewton — APPROVED** [review]
Thank you SO much for this herculean effort... it must have been tedious work, but I appreciate the standardization. :hearts: 

I just have a few minor nitpicky comments, but for the most part, this LGTM.

File: testing/dependencies/test_ants.py [hunk 3]
```diff
@@ -1,8 +1,7 @@
 #!/usr/bin/env python
-#########################################################################################
+#
 # Copyright (c) 2014 Polytechnique Montreal <www.neuro.polymtl.ca>
-# Authors: Julien Cohen-Adad
-#########################################################################################
+# License: see the file LICENSE
```
[the remark this answers is not in the document]
Comment (mguaypaq) [reply testing/dependencies/test_ants.py:4]: That makes sense.

While I'm at it, do you mind if I remove the `#!/usr/bin/env python` lines from all the testing files? They're not actually runnable scripts, so they shouldn't contain shebang lines. (And this change is more mechanical than figuring out a copyright date for each test file.)

[the remark this answers is not in the document]
Comment (joshuacwnewton) [reply testing/dependencies/test_ants.py:4]: I don't mind at all! That makes perfect sense.

**Pull request merged**